internal/server/server.go

```diff
@@ -47,11 +59,47 @@ var (
 )
 
 type Server struct {
+	// ⚡DB
 	Store storage.Store
+
+	// auth stuff (that I still dont know how to make optional)
```

Comment:
we can init a struct in go without all fields present. you have them as pointers which makes then nil by default (this is good). we can check if oauth2 is nil on startup and log a warning oauth disabled